app/src/main/java/org/simple/clinic/AppDatabase.kt

```diff
@@ -196,13 +197,13 @@ abstract class AppDatabase : RoomDatabase() {
   }
 
   @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-  fun purge() {
+  fun purge(utcClock: UtcClock) {
```

Comment:
I completely missed this before. We are supposed to use `UserClock` here and not `UtcClock`. The `retainUntil` value we save is based on the user clock, not a UTC value.